src/main/java/org/skriptlang/skript/util/ReflectUtils.java

```diff
@@ -0,0 +1,254 @@
+package org.skriptlang.skript.util;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility class for reflection.
+ */
+public class ReflectUtils {
+
+
+	public ReflectUtils() {}
+
+	/**
+	 * Cache for classes.
+	 */
+	private final Map<String, Class<?>> CLASSES = new HashMap<>();
+
+	/**
+	 * Cache for methods of classes.
+	 */
+	private final MethodMap METHODS = new MethodMap();
+
+	/**
+	 * Cache for fields of classes.
+	 */
+	private final FieldMap FIELDS = new FieldMap();
```

Comment:
these are not static and are updated after initialization, so the names should be lowercase